lib/beacon_web/plugs/put_root_layout.ex

```diff
@@ -8,7 +8,18 @@ defmodule BeaconWeb.Plug do
   end
 
   @impl Plug
-  def call(%Conn{} = conn, _) do
-    Phoenix.Controller.put_root_layout(conn, {BeaconWeb.Layouts, :root})
+  def call(%Conn{} = conn, _opts) do
+    if admin?(conn) do
+      Phoenix.Controller.put_root_layout(conn, {BeaconWeb.Layouts, :admin})
+    else
+      Phoenix.Controller.put_root_layout(conn, {BeaconWeb.Layouts, :runtime})
+    end
   end
+
+  # TODO: provide a router macro, eg: beacon_site and beacon_admin
+  defp admin?(%{private: %{phoenix_live_view: {lv, _, _}}}) do
```

Comment:
Relying on private API is definitely not ideal, I'd like to explore router macros to remove this block of code hence the `TODO`.